src/xenia/app/emulator_window.cc

```diff
@@ -174,10 +182,71 @@ bool EmulatorWindow::Initialize() {
   // FIXME: This code is really messy.
   auto main_menu = MenuItem::Create(MenuItem::Type::kNormal);
   auto file_menu = MenuItem::Create(MenuItem::Type::kPopup, L"&File");
+  // Recent Sub tap
+  auto recent_list = MenuItem::Create(MenuItem::Type::kPopup, L"&Open Recent");
+  //******Add Recent PATHS to menu***************************************
+  std::string LINES[10];
+  std::ifstream RECENT_FILE("recent.txt");
+  int i = 0;
+  // Conver from UTF-8 to whatever tf
+  std::wstring_convert<std::codecvt_utf8_utf16<wchar_t>> converter;
+  if (RECENT_FILE.is_open()) {
+    while ((i < 10) && getline(RECENT_FILE, LINES[i])) {
+      GLOBAL_RECENT_PATHS[i] = converter.from_bytes(LINES[i]);
+      i++;
+    }
+    RECENT_FILE.close();
+  }
+
+  // Added most recently opened files to menu
+  if (LINES[0] != "" && LINES[0] != "\n")
+    recent_list->AddChild(
+        MenuItem::Create(MenuItem::Type::kString, GLOBAL_RECENT_PATHS[0],
```

Comment:
This should all be a for loop. And std::bind can take static arguments, ie: `std::bind(&EmulatorWindow::RecentList, this, 0)`.

Reply:
is the last parameter (0) in 'std::bind(&EmulatorWindow::RecentList, this, 0)' passed to the function?

Reply:
Yes.

Reply:
https://en.cppreference.com/w/cpp/utility/functional/bind